paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerLecternPageChangeScriptEvent.java

```diff
@@ -0,0 +1,112 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import io.papermc.paper.event.player.PlayerLecternPageChangeEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerLecternPageChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player flips lectern page
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when the player flips to a page in a lectern.
+    //
+    // @Switch book:<item> to only process the event if the book on the lectern matches the given item.
+    //
+    // @Context
+    // <context.book> returns an ItemTag of the book in the lectern.
+    // <context.lectern> returns a LocationTag of the lectern.
+    // <context.old_page> returns an ElementTag(Number) of the last page the player was on. Page numbers follow zero-based numbering, starting at 0.
+    // <context.new_page> returns an ElementTag(Number) of the new page that the player flipped to.
+    // <context.page_direction> returns the direction in which the player flips the lectern book page, can be either LEFT or RIGHT.
+    //
+    // @Determine
+    // "PAGE:<ElementTag(Number)>" to set the page that the player will flip to.
+    //
+    // @Example
+    // # Announce the page the player flipped to.
+    // on player flips lectern page:
+    // - announce "<player.name> flipped to page #<context.new_page>!"
+    //
+    // @Example
+    // # Flips the player to page 5 if they are flagged with "pancakes".
+    // on player flips lectern page flagged:pancakes:
+    // - determine page:5
+    //
+    // -->
+
+    public PlayerLecternPageChangeScriptEvent() {
+        registerCouldMatcher("player flips lectern page");
+        registerSwitches("book");
+    }
+
+    public PlayerLecternPageChangeEvent event;
+    public LocationTag location;
+
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (!runInCheck(path, location)) {

```

Comment:
Did you test this? doesn't look like the `location` field is ever set; although in this case can probably just get the location from the event anyway